flytekit/core/annotation.py

```diff
@@ -0,0 +1,30 @@
+from typing import Any, Dict
+
+
+class FlyteAnnotation:
+    """A core object to add arbitrary annotations to flyte types.
+
+    This metadata is ingested as a python dictionary and will be serialized
+    into fields on the flyteidl type literals. This data is not accessible at
+    runtime but rather can be retrieved from flyteadmin for custom presentation
+    of typed parameters.
```

Comment:
What happens if someone annotates the return of a task with a `FlyteAnnotation`? Should we detect and raise an error in this case?